packages/react/src/hooks/useNotifications.ts

```diff
@@ -0,0 +1,123 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useNovu } from '../components/NovuProvider';
+import { Notification, NovuError, isSameFilter } from '@novu/js';
+
+export type UseNotificationsResult = {
+  readAll: () => Promise<void>;
+  archiveAll: () => Promise<void>;
+  archiveAllRead: () => Promise<void>;
+  data?: Array<Notification>;
+  error?: NovuError;
+  isLoading: boolean;
+  isFetching: boolean;
+  refetch: () => Promise<void>;
+  fetchMore: () => void;
+  hasMore: boolean;
+};
+
+export type UseNotificationsProps = {
+  tags?: string[];
+  read?: boolean;
+  archived?: boolean;
+  limit?: number;
+  onSuccess?: (data: Notification[]) => void;
+  onError?: (error: NovuError) => void;
+};
+
+export const useNotifications = (props?: UseNotificationsProps): UseNotificationsResult => {
+  const { tags, read, archived, limit, onSuccess, onError } = props || {};
+  const { notifications, on } = useNovu();
+  const [data, setData] = useState<Array<Notification>>();
```

Comment:
@desiprisg This solution works with the internal useState works. Did you consider using https://react.dev/reference/react/useSyncExternalStore instead and make it listen to the Novu events? 

I feel like it might help us with the `setData` handling as it's a hook tailored to be used with underlying event-driven state managers like the Novu class.

Reply:
will try it out

Reply:
Tbh I like the other approach more, since we have to call list() anyway in a useEffect. It also acts as a more universal example for other framework implementations.